### ISSUE TITLE
Ref is undefined when not populating

### ISSUE THREAD

**Issue by mikheevm** [typegoose/typegoose#178]
## Versions

- NodeJS: 10.16.3
- Typegoose(NPM): 6.2.1 
- mongoose: 5.8.9
- mongodb: 4.2.1
- nestjs / nestjs-typegoose: 6.11.0 / 7.0.0 

## What is the Problem?

A field with reference is undefined until you populate it

## Code Example

Token Model:
```ts
export class VerificationToken {
  @IsString()
  @prop({
    ref: User,
    required: true,
  })
  user!: Ref<User>;

  @IsString()
  @prop({
    required: true,
  })
  token!: string;

  @IsDate()
  @prop({
    required: true,
    default: Date.now,
    expires: 43200,
  })
  createdAt!: Date;
}
```

User Model:
```ts
export class User {
  @IsString()
  @prop({ default: () => nanoid() })
  _id!: string;
}
```

Usage:
```
const storedToken = await this.verificationTokenModel.findOne({ token });

console.log(token.user) // undefined
```

If you call it with `populate('user')` everything is ok

## Do you know *why* it happenes?

*no*

**Comment by hasezoey** [comment]
@mikheevm and how did you create the verificationTokenModel document that is to be found? does it have an id assigned to the property `user`? otherwise i dont see a problem with it

**Comment by mikheevm** [comment]
@hasezoey 
```ts
     const user = new this.userModel(userData);
     await user.save();

    const tokenString = crypto.randomBytes(16).toString('hex');
    const verificationToken = new this.verificationTokenModel({
      user,
      token: tokenString,
    });
```

**Comment by hasezoey** [comment]
sorry, cant verify, used code:

```ts
// NodeJS: 12.14.1
// MongoDB: 4.2-bionic (Docker)
import { getModelForClass, prop, Ref, isDocument } from "@typegoose/typegoose"; // @typegoose/typegoose@6.2.1
import * as mongoose from "mongoose"; // mongoose@5.8.7

class User {
  @prop()
  public username?: string;
}
const UserModel = getModelForClass(User);

class VerificationToken {
  @prop({
    ref: User,
    required: true
  })
  public user!: Ref<User>;
}
const VerificationTokenModel = getModelForClass(VerificationToken);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verify178", useCreateIndex: true, useUnifiedTopology: true });

  const userdoc = new UserModel({ username: "user1" });
  console.log(userdoc);
  await userdoc.save();

  const vtdoc = new VerificationTokenModel({ user: userdoc });
  console.log(vtdoc);
  await vtdoc.save();

  const found = await VerificationTokenModel.findOne({}).exec();
  if (isDocument(found)) { // ensure it is actually found (satisfy typescript)
    console.log(found);
    console.log(found.user);
  } else {
    throw new Error("VT Not found!");
  }

  await mongoose.disconnect();
})();
```

prints (as it should): 
```
{ _id: 5e2b5666f1e7225922c8783e, username: 'user1' } // the created user
{ // the created VerificationToken
  _id: 5e2b5666f1e7225922c8783f,
  user: { _id: 5e2b5666f1e7225922c8783e, username: 'user1', __v: 0 }
}
{ // the full found VerificationToken
  _id: 5e2b5666f1e7225922c8783f,
  user: 5e2b5666f1e7225922c8783e,
  __v: 0
}
5e2b5666f1e7225922c8783e // the value of "doc.user"
```

**Comment by mikheevm** [comment]
i'm using nano id (not the ObjectId) could this be the reason?

**Comment by hasezoey** [comment]
theoretically it shouldnt affect it, but there are known instances where it fails with custom types in typegoose, i will need to check nanoid, then i will report back

**Comment by hasezoey** [comment]
@mikheevm fixed in 6.2.2, didnt notice the use of `rawOptions.type` after deletion of it

PS: you will still need to set the Ref-type to string:

```ts
export class VerificationToken {
  @prop({
    ref: User,
    required: true,
    type: String // for mongoose
  })
  user!: Ref<User, string>; // for the types
}
```

**Comment by mikheevm** [comment]
@hasezoey Thank you!
